ra/ra.go

```diff
@@ -815,6 +815,10 @@ func (ra *RegistrationAuthorityImpl) FinalizeOrder(ctx context.Context, req *rap
 		return nil, err
 	}
 
+	if err := csrlib.VerifyCSR(csrOb, ra.maxNames, &ra.keyPolicy, ra.PA, ra.forceCNFromSAN, *req.Order.RegistrationID); err != nil {
```

Comment:
Oops, I was mixing up `ra.issueCertificate` and `ra.NewCertificate`. I think probably the call to `VerifyCSR` in `ra.issueCertificate` should be moved into `ra.NewCertificate`. It's needed in `ra.FinalizeOrder` because that method does a number of checks for the order matching the CSR. The one that caused this issue was the one right below (L828) which checks if `csrOb.DNSNames == order.Names` and since `csr.VerifyCSR` is (probably wrongly) responsible for moving any CN to the DNSNames this would fail with a mixed CSR.

Probably the easiest solution to avoid duplicating calls to `csr.VerifyCSR` is to simply move the call in `ra.issueCertificate` out into `ra.NewCertificate` just before it calls the former. Thoughts?

Reply:
> Probably the easiest solution to avoid duplicating calls to csr.VerifyCSR is to simply move the call in ra.issueCertificate out into ra.NewCertificate just before it calls the former. Thoughts?

I think that makes sense :+1:

Reply:
If the only reason this needs to be called here is that `VerifyCSR` is (wrongly) responsibly for hoisting a SAN into CN, why not change that behavior instead? Bear in mind that `forceCNFromSAN` is not launching anytime soon (needs a ballot, and the implementation also need changing), so I think it's worthwhile to remove it; either separately, if that's easier, or as part of this PR, if it makes this PR easier.

Reply:
I think moving this up higher actually makes sense regardless of the side-effect we are fixing here, it allows us to avoid a bunch of extra work that we shouldn't be doing if the CSR is obviously malformed in some way anyway.

Note: this has nothing to do with `forceCNFromSAN`, it's actually relying on the opposite behaviour, adding the CN to the SAN if it's missing.

Reply:
> Note: this has nothing to do with forceCNFromSAN, it's actually relying on the opposite behaviour, adding the CN to the SAN if it's missing.

Well, my thinking here was that the reason that side-effect is done in `VerifyCSR` is *probably* because it depends on the value of `forceCNFromSAN`, so hoisting it out would probably involve removing that dependency. At any rate, I'll review as-is.